diff --git a/crypto/fipsmodule/ec/p256_test.cc b/crypto/fipsmodule/ec/p256_test.cc
index a8039c385..15eabb046 100644
--- a/crypto/fipsmodule/ec/p256_test.cc
+++ b/crypto/fipsmodule/ec/p256_test.cc
@@ -12,20 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <assert.h>
 #include <gtest/gtest.h>
 #include "../../internal.h"
 #include "../../test/abi_test.h"
 
-
-BSSL_NAMESPACE_BEGIN
-namespace {
-
 #if !defined(OPENSSL_NO_ASM) && defined(__GNUC__) && defined(__x86_64__) && \
     defined(SUPPORTS_ABI_TEST)
 extern "C" {
 #include "../../../third_party/fiat/p256_field.c.inc"
 }
 
+BSSL_NAMESPACE_BEGIN
+namespace {
+
 TEST(P256Test, AdxMulABI) {
   static const uint64_t in1[4] = {0}, in2[4] = {0};
   uint64_t out[4];
@@ -37,7 +37,6 @@ TEST(P256Test, AdxMulABI) {
   }
 }
 
-#include <assert.h>
 TEST(P256Test, AdxSquareABI) {
   static const uint64_t in[4] = {0};
   uint64_t out[4];
@@ -48,7 +47,8 @@ TEST(P256Test, AdxSquareABI) {
     GTEST_SKIP() << "Can't test ABI of ADX code without ADX";
   }
 }
-#endif
 
 }  // namespace
 BSSL_NAMESPACE_END
+
+#endif
